src/auth/baseexternalclient.ts

```diff
@@ -169,9 +160,10 @@ export abstract class BaseExternalAccountClient extends AuthClient {
    */
   constructor(
     options: BaseExternalAccountClientOptions,
-    additionalOptions?: RefreshOptions
+    additionalOptions?: AuthClientOptions
```

Comment:
This seems very confusing to me now that the BaseExternalAccountClientOptions object is a child of AuthClientOptions, doesn't seem like it will be clear to users what should be where. It would make sense to me to create a new constructor that takes in just 1 options objects (BaseExternalAccountClientOptions) and does the existing constructor logic, and then keep this constructor for backwards compatibility but just have it pass the merged options objects to the new constructor like you are calling the super here. And then have comments explaining that the old constructor is deprecated/only for compatability

Reply:
I wish I was around for the naming 😅 `BaseExternalAccountClient` is the base of external, non-Google Cloud accounts (AWS, Azure, etc.). Since all `AuthClient`s are required to have `transporter`, `projectId`, `quotaProjectId`, and now `universeDomain` properties it would make since for this to be handled in the root class.

Reply:
Just realized we can't have multiple constructors in typescript so what I was suggesting doesn't make sense anyway. I agree that it make sense to handle it all in the base class. My concern is that having both options objects will be confusing to users since they are both AuthClientOptions. At the very least can you add to the constructor documentation that the additional options param is deprecated and the options can all be passed via the BaseExternalAccountClientOptions parameter? Other option is if we just wait to release this until the breaking change suggested by  #1672 then we won't be in this weird in between state at all.

Reply:
Docs-deprecated `additionalOptions`, as the [@deprecated](https://tsdoc.org/pages/tags/deprecated/) tag is considered breaking as it would produce build warnings.